modules/nf-core/csvtk/join/main.nf

```diff
@@ -0,0 +1,55 @@
+process CSVTK_JOIN {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::csvtk=0.26.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/csvtk:0.26.0--h9ee0642_0':
+        'biocontainers/csvtk:0.26.0--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(csv)
+    val in_format
+    val out_format
+
+    output:
+    tuple val(meta), path("${prefix}.${out_extension}"), emit: csv
+    path "versions.yml"                                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def delimiter = in_format == "tsv" ? "\t" : (in_format == "csv" ? "," : in_format)
+    def out_delimiter = out_format == "tsv" ? "\t" : (out_format == "csv" ? "," : out_format)
+    out_extension = out_format == "tsv" ? 'tsv' : 'csv'
```

Comment:
```suggestion
    out_extension = args.contains('--out-delimiter "\t"') || args.contains('-D "\t"' ? "tsv" : "csv"
```
This looks if `--out-delimiter` or `-D` is in the args. If `\t` is the option it will use `tsv` as the output extension, otherwise it will use `csv` (which is the default delimiter)